CollapseLauncher/Classes/RegionManagement/RegionManagement.cs

```diff
@@ -298,11 +299,27 @@ internal static async ValueTask TryDownloadToCompleteness(string url, FileInfo f
             }
         }
 
+        private string GetDeviceId(PresetConfigV2 Preset)
+        {
+            var deviceId = (string)Registry.GetValue(Preset.InstallRegistryLocation, "UUID", null);
+            if (deviceId == null)
+            {
+                const string regKeyCryptography = "HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Cryptography";
+                var guid = (string)Registry.GetValue(regKeyCryptography, "MachineGuid", null) ?? Guid.NewGuid().ToString();
+                deviceId = guid.Replace("-", "") + (long)DateTime.Now.Subtract(DateTime.UnixEpoch).TotalMilliseconds;
+                Registry.SetValue(Preset.InstallRegistryLocation, "UUID", deviceId);
+            }
+            return deviceId;
+        }
+
         private async ValueTask FetchLauncherDownloadInformation(CancellationToken token, PresetConfigV2 Preset)
         {
             _gameAPIProp = await FallbackCDNUtil.DownloadAsJSONType<RegionResourceProp>(Preset.LauncherResourceURL, InternalAppJSONContext.Default, token);
             if (!string.IsNullOrEmpty(Preset.LauncherPluginURL))
             {
+                // TODO: use string.Format
+                Preset.LauncherPluginURL = Preset.LauncherPluginURL.Replace("111111111111111111111111111111111111111111111", GetDeviceId(Preset));
+
```

Comment:
@bagusnl Due to this change, we need to replace the ``111111111111111111111111111111111111111111111`` placeholder to ``{0}`` so we can do the formatting to put the GUID instead of replacing it.

@shatyuka You can change the code to use string.format so we will change the GUID placeholder with ``{0}`` in our metadata.

Reply:
was just gonna ask about it LOL 
on it o7